lib/Doctrine/Migrations/Configuration/AbstractFileConfiguration.php

```diff
@@ -141,9 +141,9 @@ private function loadMigrations(array $migrations) : void
 
     private function setMigrationOrganization(string $migrationOrganization) : void
     {
-        if (strcasecmp($migrationOrganization, static::VERSIONS_ORGANIZATION_BY_YEAR) === 0) {
+        if (strcasecmp($migrationOrganization, self::VERSIONS_ORGANIZATION_BY_YEAR) === 0) {
```

Comment:
```
  97 | ERROR | [x] Late static binding for constants is disallowed.
     |       |     (SlevomatCodingStandard.Classes.DisallowLateStaticBindingForConstants.DisallowedLateStaticBindingForConstant)
```

🤔 - thinking of it, this is not so innocent. Should we force the change here or should we rather exclude the rule for this file?

Reply:
What is the issue? The change seems fine unless I'm missing something.

Reply:
Due to the fact it's a abstract class and intended to be extended, when switching from `static::VERSIONS_ORGANIZATION_BY_YEAR ` to `self::VERSIONS_ORGANIZATION_BY_YEAR ` we change behavior here.

With `static::VERSIONS_ORGANIZATION_BY_YEAR ` the constant from the subClass would be used - whereas with `self::VERSIONS_ORGANIZATION_BY_YEAR ` - the constant from `AbstractFileConfiguration` would always be used ( from my understanding ).

Please correct me if I am wrong

Reply:
I understand what you mean now. I don't think it matters in practice. These classes are internal and not intended for the end user to extend so the constant should not ever be overridden.